File: 2793-count-the-number-of-complete-components/count-the-number-of-complete-components.java
class Solution {
    public int countCompleteComponents(int n, int[][] edges) {
        List<List<Integer>> list = new ArrayList<>();
        for(int i=0; i<n; i++) list.add(new ArrayList<>());
        for(int i=0; i<edges.length; i++){
            list.get(edges[i][0]).add(edges[i][1]);
            list.get(edges[i][1]).add(edges[i][0]);
        }
        boolean visited[] = new boolean[n];
        int res= 0;
        System.out.println(list);
        for(int i=0; i<n; i++){
            if(!visited[i]){
                Queue<Integer> q = new LinkedList<>();
                q.add(i);
                visited[i] = true;
                int size = list.get(i).size()+1;
                int count = 1;
                boolean complete = true;
                while(!q.isEmpty()){
                    int temp = q.poll();
                    for(int x:list.get(temp)){
                        if(!visited[x]){
                            count++;
                            if(list.get(x).size()+1!=size) complete = false;
                            q.add(x);
                            visited[x] = true;
                        }
                    }
                }
                if(size!=count) complete = false;
                if(complete) res++;
            }
        }
        return res;
    }
}

File: 2793-count-the-number-of-complete-components/count-the-number-of-complete-components.cpp
class Solution {
public:
    int countCompleteComponents(int n, vector<vector<int>>& edges) {
        map<int,vector<int>> map;
        for(auto i: edges){
            map[i[0]].push_back(i[1]);
            map[i[1]].push_back(i[0]);
        }
        int res =0;
        vector<bool>visited(n,false);
        for(int i=0; i<n; i++){
            if(visited[i]!=true){
                queue<int> q;
                q.push(i);
                bool flag = true;
                int req = map[i].size();
                int count =0;
                while(!q.empty()){
                    int temp = q.front();
                    q.pop();
                    if(map[temp].size()!=req) flag = false;
                    visited[i] = true;
                    for(auto x:map[temp]){
                        if(!visited[x]){
                            count++;
                            visited[x] = true;
                            q.push(x);
                        }
                    }
                }
                if(flag && count==req) res++;
            }
        }
        return res;
    }
};